### ISSUE TITLE
bug: neotest for neotest-dotnet nto working

### ISSUE THREAD

**Issue by Tantol** [LazyVim/LazyVim#2348]
### Did you check docs and existing issues?

- [X] I have read all the LazyVim docs
- [X] I have searched the existing issues of LazyVim
- [X] I have searched the existing issues of plugins related to this issue

### Neovim version (nvim -v)

0.10.0

### Operating system/version

Ubuntu 23.04

### Describe the bug

"No test found" when running nearest test.

### Steps To Reproduce

```
return {
  { "Issafalcon/neotest-dotnet" },
  {
    "nvim-neotest/neotest",
    opts = { adapters = { "neotest-dotnet" } },
  }
}
  ```

### Expected Behavior

Running nearest .Net tests with neotest-deotnet.

_Or some help with setting it up with any plugin to run and debug tests in .Net_

**Comment by Tantol** [comment]
I manage to fix my issue by adding `nvim-treesitter/nvim-treesitter` as dependency when changing `nvim-neotest/neotest` config.

Working configuration with debugging functionality:
```
return {
  {
    "nvim-neotest/neotest",
    dependencies = {
      {
        "issafalcon/neotest-dotnet",
        opts = {
          custom_attributes = {
            xunit = { "TestScenarios" }
          }
        },
        config = function()
          local dap = require('dap')

          if not dap.adapters["netcoredbg"] then
            local mason_path = vim.fn.glob(vim.fn.stdpath "data" .. "/mason/")
            dap.adapters.netcoredbg = {
              type = 'executable',
              command = mason_path .. "bin/netcoredbg",
              args = {'--interpreter=vscode'}
            }
          end
        end,
      },
      "nvim-treesitter/nvim-treesitter",
    },
    opts = {
      adapters = { "neotest-dotnet" }
    }
  },
}
```

Still it's kinda buggy to remember to add `treesitter` dependency. Doc example is missing that: https://www.lazyvim.org/extras/test/core

**Comment by dpetka2001** [comment]
@Tantol The docs example is missing this piece of information, because it was added last [month](https://github.com/nvim-neotest/neotest/commit/b8e29c0fba9a58bf6a5c37df77c7a6a31079c8d6) on the `neotest` repo itself. Initially they required `nvim-treesitter` either in the neotest plugin itself or in the neotest adapters (I don't remember which). They dropped this support for their reasons and now `nvim-treesitter` has to be required as a dependency. Lazyvim docs about `neotest` were initially compiled 8 [months](https://github.com/LazyVim/lazyvim.github.io/blame/main/docs/extras/test/core.md) ago. So, obviously this was not the case at the time. 

Understandably, the docs have fallen behind at the moment until the maintainer comes back to correct them according to the current state of things. But this was not done, in my opinion as a LazyVim user always who has been using Lazyvim for about a year now, because something was missed by the maintainer at the time the docs were made.

Simply put the state of things has changed and I agree with you that the docs have to also reflect this change. But it's not a LazyVim bug according to my opinion.
